### PR TITLE
Add documentation for V4 Signature Post Policy

### PR DESCRIPTION
**Pull request by frankyn**

### CODE REVIEW

File: storage/v1/proto/google/cloud/conformance/storage/v1/tests.proto [hunk 1]
```diff
@@ -75,6 +79,40 @@ message PolicyInput {
 message PolicyOutput {
   string url = 1;
   map<string, string> fields = 2;
+  /*
+    Expectations
+
+    E.1: PolicyInput.fields must be preppended to form expectedDecodedPolicy
```
Comment (tritone) [inline storage/v1/proto/google/cloud/conformance/storage/v1/tests.proto:85]: sp: prepended

Also, prepended to what? From the example it looks like you're saying that these should come before the other conditions in the policy but it's not 100% clear.

File: storage/v1/proto/google/cloud/conformance/storage/v1/tests.proto [hunk 2]
```diff
@@ -75,6 +79,40 @@ message PolicyInput {
...
+    for consistent result across languages. Ordering doesn't matter to the
+    service but the decision is made to make it easier to conform implementations
+    in implementation.
+
+    Example:
+    PolicyInput.fields has:
+    {
+      "content-disposition":"attachment; filename=\"~._-%=/é0Aa\"",
```
Comment (tritone) [inline storage/v1/proto/google/cloud/conformance/storage/v1/tests.proto:93]: Is this required to be sorted alphabetically by key? If so can that be stated explicitly?

File: storage/v1/proto/google/cloud/conformance/storage/v1/tests.proto [hunk 3]
```diff
@@ -75,6 +79,40 @@ message PolicyInput {
...
+      "content-encoding":"gzip",
+      "content-type":"text/plain",
+      "success_action_redirect":"http://www.google.com/"
+    }
+
+    Then expectedDecodedPolicy should preppends PolicyInput.fields in
```
Comment (tritone) [inline storage/v1/proto/google/cloud/conformance/storage/v1/tests.proto:99]: sp: prepend

File: storage/v1/proto/google/cloud/conformance/storage/v1/tests.proto (hunk 1 above)
Comment (frankyn) [reply storage/v1/proto/google/cloud/conformance/storage/v1/tests.proto:85]: That's correct. I'll update the wording.

File: storage/v1/proto/google/cloud/conformance/storage/v1/tests.proto (hunk 2 above)
Comment (frankyn) [reply storage/v1/proto/google/cloud/conformance/storage/v1/tests.proto:93]: No, the ordering just needs to match the conformance test input.

**Review by tritone — APPROVED** [review]
Thanks for adding this Frank!

**Pull request merged**